### PR TITLE
Distraction free mode: Fix keyboard shortcut not working

### LINKED ISSUE #47873: Distraction free mode: Keyboard shortcut not working
### Description

Related to: #41740

Pressing the shortcut key indicated in the options menu does not switch to the distraction-free mode.

Shortcuts to switch to full-screen mode or to the code editor both work correctly. I tested on Windows, but would be happy if someone could test to see if this problem reproduces on a Mac as well.

### Step-by-step reproduction instructions

- Open the editor.
- Press `Ctrl` + `Shift` + `\` key (This is a shortcut for Windows) .
- Nothing happens

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/54422211/217525824-aa96dc73-d2b9-44c0-81eb-8d60d9906b53.mp4

### Environment info

- OS: Windows 11
- Wordpress Version: WordPress 6.2-alpha-55250
- Gutenberg Version: Version 15.1.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes: #47873
Related to: #43428

## What?
This PR fixes a problem where the shortcut for switching distraction flow mode did not work (probably Windows only).

## Why?

The shortcut for switching the distraction free mode is as follows:

- Windows: `Ctrl`+`Shift`+`\`
- Mac: `⇧`+`⌘`+`\`

The `event.key` for determining shortcuts has a value that takes into account modifier keys such as the `shift` key. Therefore, as I have investigated in this comment, I believe the cause is that when the `shift` key is combined with other keys, `event.key` has a value that is not originally expected. Furthermore, the character changed by the `shift` key depends on the keyboard layout, so `event.key` cannot be used as the source of the decision.

## How?

Refers to the value of `event.code` that represents a physical key on the keyboard. Based on that value, it is converted to the originally expected `event.key`.

This PR fixes the bug in Windows. However, as mentioned in [the two comments from here](https://github.com/WordPress/gutenberg/issues/47873#issuecomment-1423175792), the test results seem to be different for Mac. I would appreciate it if you could test to see if this shortcut issue is currently occurring on Mac.

### Testing Instructions for Keyboard

- Confirm that this shortcut key works in the post editor.
- Confirm that the other shortcuts also work as before.

### English Keyboard Layout

`event.code` is `Backslash`.

https://user-images.githubusercontent.com/54422211/217984887-e7fb7f37-2242-4b7e-989e-a2cade0e56be.mp4

### Japanese Keyboard Layout

`event.code` is `IntlRo` or `IntlYen`.

https://user-images.githubusercontent.com/54422211/217984898-dcf49982-636d-430c-ae33-ab28575f4d03.mp4

### CODE REVIEW

File: packages/keycodes/src/index.js [hunk 1]
```diff
@@ -390,6 +390,14 @@ export const isKeyboardEvent = mapValues(
 		) => {
 			const mods = getModifiers( _isApple );
 			const eventMods = getEventModifiers( event );
+			/** @type {Record<string,string>} */
+			const replacementWithShiftKeyMap = {
+				Comma: ',',
+				Backslash: '\\',
+				// Windows returns `\` for both IntlRo and IntlYen.
```
Comment (t-hamano) [inline packages/keycodes/src/index.js:397]: This comment was taken from [the source code of chromium](https://chromium.googlesource.com/chromium/src/+/lkgr/ui/events/keycodes/dom/dom_keyboard_layout_map_win.cc#110)

**Review by alexstine — APPROVED** [review]
@t-hamano Works fine for me.

**Comment by t-hamano** [comment]
Thank you for the review, @alexstine!

I will wait for someone to test it on Mac (because I don't have Mac...).

**Comment by alexstine** [comment]
@t-hamano I have Mac as well, what is the keyboard shortcut? Spell it out if you do not mind, I cannot figure out what the icons mean in the PR description.

**Comment by t-hamano** [comment]
Thank you for the test, @alexstine.

The shortcut on the Mac would be ‘shift‘ + `command` + 'backslash'. This PR is for Windows only, but it would be appreciated if you could confirm if this problem originally occurred on Mac as well 🙏

**Comment by alexstine** [comment]
@t-hamano Yes, this issue does occur on Mac for trunk. It also is not impacted by your PR. Pressing cmd+shift+\ results in opening the new tabs in Safari. Seems like event is not prevented or the shortcut is not getting interpreted correctly.

**Comment by t-hamano** [comment]
Thanks for the test, @alexstine!

I have removed `isApple()` so that this PR applies to Macs as well. However, this change may also affect shortcuts for opening and closing the sidebar on Mac.

Shortcut key is  `shift` + `command` + `, (Comma)`. It would be great if you could also test if this shortcut works as well on Macs as it has in the past 🙏

**Review by ntsekouras — APPROVED** [review]
>Shortcut key is shift + command + , (Comma). It would be great if you could also test if this shortcut works as well on Macs as it has in the past 🙏

Tested and works before. Thanks @t-hamano !

**Comment by ntsekouras** [comment]
Cherry-picked this PR to the wp/6.2 branch.

**Pull request merged**